src/mds/ScrubStack.cc

```diff
@@ -299,6 +302,24 @@ void ScrubStack::scrub_dirfrag(CDir *dir,
   *is_terminal = false;
   *done = false;
 
+  if (!dir->is_auth() && dir->is_subtree_root()) {
```

Comment:
Because the assert(is_auth) in enqueue_scrub_work went away, couldn't we end up in here with a dir that we are not auth for, but is also not a subtree root?  In that case we would want to send off a MMDSScrubPath as well.